src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -0,0 +1,99 @@
+
+        static void Main(string[] args)
+        {
+            //Setting a default value
+            Level level = Level.Null;
+
+            using (var parser = new CommandLine.Parser(with => with.HelpWriter = Console.Error))
+            {
+                if (parser.ParseArgumentsStrict(args, _options, () => Environment.Exit(-1)))
+                {
+                    if (_options.Help)
+                    {
+                        Console.WriteLine(CommandLine.Text.HelpText.AutoBuild(_options));
+                        return;
+                    }
+                   
+                    if(_options.Output != null)
+                    {
+                        if (!Directory.Exists(_options.Output))
+                        {
+                            Directory.CreateDirectory(_options.Output);
+                            Console.WriteLine("created directory at {0}", _options.Output);
+                        }
+                    }
+
+                    if (_options.Level != null)
+                    {
+                        try
+                        {
+                            level = (Level)Enum.Parse(typeof(Level), _options.Level, true);
+                            WriteToTxtFile("file", _options.Output, _options.DLLs, level);
+                        }
+                        catch (ArgumentException e)
+                        {
+                            Console.WriteLine("ArgumentException: Please enter either Namespace, Class, or Function for level");
+                        }
+                    }
+                }
+            }
+        }
+        //While reflection is not implemented this method writes path of dll if it exists with specified level to txt file
+        private static void WriteToTxtFile(string name, string outputDirectory, IList<string> dllDirectories, Level level)
+        {
+            //user entered some dlls
+            if (dllDirectories.Count != 0)
+            {
+                if (outputDirectory == null)
+                {
+                    Console.WriteLine("No output directory specified, writing to stdout");
+                    foreach (string dll in dllDirectories)
+                    {
+                        if (File.Exists(dll))
+                        {
+                            Console.Out.WriteLine("{0}|{1}", dll, level);
+                        }
+                        else
+                        {
+                            Console.Out.WriteLine("Error: {0} file cannot be found on system", dll);
+                        }
+                    }
+                }
+                else
+                {
+                    foreach (string dll in dllDirectories)
+                    {
+                        if (File.Exists(dll))
+                        {
+                            File.AppendAllText(outputDirectory + name + ".txt", dll + "|" + level + Environment.NewLine);
```

Comment:
Instead of having separate if/else blocks for writing to stdout vs a file, you can use a StreamWriter and pass it either Console.Out or the output filename.  See:

https://msdn.microsoft.com/en-us/library/system.io.streamwriter(v=vs.110).aspx
and
https://msdn.microsoft.com/en-us/library/system.console.out(v=vs.110).aspx

Then you can write to stdout and files in the same way.  Plus then you'd only need to open the file once; write all the lines and then close it at the end.